### PR TITLE
[CDAP-7465] Have consistent view for packages in Marketplace

### PR DESCRIPTION
**Pull request by ajainarayanan**
- Manually overlays details of a package in the marketplace
- Handles the case to have open entity opened at any time.
- Adds generic ActionsContainer to be reused across all packages.

Bamboo build: http://builds.cask.co/browse/CDAP-DRC4762-6

### CODE REVIEW

File: cdap-ui/app/cdap/components/MarketActionsContainer/index.js [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+          this.props
+            .actions
+            .map((action, index) => {
+              let isCompletedAction = this.state.completedActions.indexOf(index) !== -1 ;
+              let actionName = T.translate('features.Market.action-types.' + action.type + '.name');
+              let actionIcon = getIcon(action.type);
+              return (
+                <div
+                  className="action-container text-center"
+                  key={shortid.generate()}
```
Comment (elfenheart) [inline cdap-ui/app/cdap/components/MarketActionsContainer/index.js:84]: Will this cause a problem by re-rendering all the child element every time the render() is called?

File: cdap-ui/app/cdap/components/MarketPlaceEntity/MarketPlaceEntity.less [hunk 2]
```diff
@@ -15,39 +15,130 @@
...
+      width: ~"-moz-calc(100% - 145px)";
+
+      > .package-label {
+        font-size: 17px;
       }
+      .package-metadata {
+        margin-top: 10px;
+      }
+    }
+    .pacakge-footer {
```
Comment (elfenheart) [inline cdap-ui/app/cdap/components/MarketPlaceEntity/MarketPlaceEntity.less]: typo?

File: cdap-ui/app/cdap/components/MarketPlaceEntity/index.js [hunk 3]
```diff
@@ -13,35 +13,255 @@
...
+                onClick={this.toggleDetailedMode.bind(this)}>
+                <img src={MyMarketApi.getIcon(this.props.entity)} />
+              </div>
+
+              <div className="package-medata-container">
+                <strong className="package-label"> {this.props.entity.label} </strong>
+                <div className="package-metadata">
+                  <div>
+                    <span>
+                      <strong> Version </strong>
```
Comment (elfenheart) [inline cdap-ui/app/cdap/components/MarketPlaceEntity/index.js]: i18n?

**Comment by PGuimarais** [comment]
Spoke with @ajainarayanan , couple minor changes:
- When expanding a marketplace entity card, the primary icon shifts a few pixels over. Change will be to have it remain in place when expanding / collapsing card.
- Would be good to have esc key collapse a currently opened card, rather than having the esc key close the underlying marketplace modal

File: cdap-ui/app/cdap/components/MarketActionsContainer/index.js (hunk 1 above)
Comment (ajainarayanan) [reply cdap-ui/app/cdap/components/MarketActionsContainer/index.js:84]: It shouldn't be a problem. We are rendering this only once per package. Clicking any other package should close this. So generating IDs inside a map for a one time rendering list shouldn't be a problem.

**Comment by elfenheart** [comment]
👍  LGTM

**Pull request merged**